roles/rsyslog/README.md

```diff
@@ -269,17 +280,22 @@ Viaq sub-variables
 ------------------
 - `rsyslog_viaq_config_dir`: Directory to store viaq configuration files.  Default to '/etc/rsyslog.d/viaq'.
 - `rsyslog_viaq_log_dir`: Viaq log directory.  Default to '/var/log/containers'.
-- `openshift_logging_use_ops`: Set to 'True', if you have a second ES cluster for infrastructure logs. Default to 'False'.
 - `logging_mmk8s_token`: Path to token for kubernetes.  Default to "/etc/rsyslog.d/viaq/mmk8s.token"
 - `logging_mmk8s_ca_cert`: Path to CA cert for kubernetes.  Default to "/etc/rsyslog.d/viaq/mmk8s.ca.crt"
-- `rsyslog_elasticsearch_viaq`: A set of following variables to specify output elasticsearch configurations.  It could be an array if multiple elasticsearch clusters to be configured. 
+- `rsyslog_elasticsearch`: A set of following variables to specify output elasticsearch configurations. It could be an array if multiple elasticsearch clusters to be configured.
+- `use_omelasticsearch_cert` : If set to `True`, omelasticsearch is configured to use the certificates specified in rsyslog_elasticsearch.  Default to `False`.
```

Comment:
The CA cert is required if using `usehttps`: https://www.rsyslog.com/doc/v8-stable/configuration/modules/omelasticsearch.html#usehttps - so I suppose we need to provide an ansible parameter for `usehttps`.

In addition - if using client cert auth, you must have `usehttps="on"`, and you must specify a client cert and key

Reply:
It is part of the action parameters. For viaq the default id "on".
I was thinking how to generelize this for any omelasticsearch output where user can just pass the plugin parameters to the "action". I still dont have a solution that works.
@richm @nhosoi @pcahyna

Reply:
Indeed, this is a bug.  Currently, 'usehttps="on"' is hardcoded in the defaults/main.yaml.  (there are 4 of them.)  Can we move usehttps="on" in the if use_omelasticsearch_cert clause (exchanging line 992 and 993)?  And the client auth configuration needs to be added...
```
 977               action(
 978                   type="omelasticsearch"
 979                   name="{{ res.name | default('viaq-elasticsearch') }}"
                          ..........................................
 992                   usehttps="on"
 993           {% if use_omelasticsearch_cert | default(False) %}
 994                   tls.cacert="{{ res.ca_cert }}"
 995                   tls.mycert="{{ res.cert }}"
 996                   tls.myprivkey="{{ res.key }}"
 997           {% endif %}
 998               )
```

Reply:
> Can we move usehttps="on" in the if use_omelasticsearch_cert clause (exchanging line 992 and 993)?

For a general purpose elasticsearch configuration api, we need a separate option for `usehttps="on"` - maybe `use_omelasticsearch_https` - then add to defaults.yaml - `use_omelasticsearch_https: {{ true if use_omelasticsearch_cert else false }}` - not sure about the ansible syntax here , but the gist is that you want `use_omelasticsearch_https` to default to `true` if the user has set `use_omelasticsearch_cert: true`, otherwise, `false`.  Then something like this:
```
   usehttps="{{ 'on' if use_omelasticsearch_https else 'off' }}"
```
again - not sure of the exact ansible syntax here

Reply:
so, if I understand correctly, if "use_omelasticsearch_cert" is "true" then "usehttps" is "on".
Is there an option to use "usehttps" is "on" without the tls certs?
@richm

Reply:
Probably, we should distinguish encrypting the communication (usehttps on) from setting up certs?  We cannot enable usehttps without certs, but we can disable it even if the certs are configured.

Reply:
You can use omelasticsearch without certs and without https.

You can set `usehttps="on"` if
- you set `allowunsignedcerts="on"` (default `"off"` - insecure to use `"on"` - not recommended)
OR
- you specify a ca cert with `tls.cacert`

You can use client cert auth if
- you set `usehttps="on"`
AND
- you provide a CA cert with `tls.cacert`
AND
- you provide both `tls.mycert` and `tls.myprivkey`

> Probably, we should distinguish encrypting the communication (usehttps on) from setting up certs?

Yes, but you cannot use the client cert/key if `usehttps="off"`

> We cannot enable usehttps without certs,

You can - see above - but having a ca cert is strongly encouraged and secure.

> but we can disable it even if the certs are configured.

I suppose there are cases where the user may want to provide certs, but not use `https`?  But then the certs are useless.

Reply:
Is allowunsignedcerts a parameter in the omelasticsearch?  I cannot find it in this doc...
https://www.rsyslog.com/doc/v8-stable/configuration/modules/omelasticsearch.html

The mmkubernetes module has it, though...
https://www.rsyslog.com/doc/v8-stable/configuration/module/mmkubernetes.html#allowunsignedcerts

Reply:
Lets open a separate issue on this and track it there. ok?
@nhosoi @richm

Reply:
> Is allowunsignedcerts a parameter in the omelasticsearch? I cannot find it in this doc...
> https://www.rsyslog.com/doc/v8-stable/configuration/modules/omelasticsearch.html

It is not documented: https://github.com/rsyslog/rsyslog/blob/master/plugins/omelasticsearch/omelasticsearch.c#L203

> 
> The mmkubernetes module has it, though...
> https://www.rsyslog.com/doc/v8-stable/configuration/module/mmkubernetes.html#allowunsignedcerts

Reply:
Thanks, @richm.  Indeed, it is in the source code.